src/features/utils/instrument-base.js

```diff
@@ -50,6 +50,15 @@ export class InstrumentBase extends FeatureBase {
     if (getConfigurationValue(this.agentIdentifier, `${this.featureName}.autoStart`) === false) this.auto = false
     /** if the feature requires opt-in (!auto-start), it will get registered once the api has been called */
     if (this.auto) registerDrain(agentIdentifier, featureName)
+    else {
+      this.ee.on(`${this.featureName}-opt-in`, () => {
+        // register the feature to drain only once the API has been called, it will drain when importAggregator finishes for all the features
+        // called by the api in that cycle
+        registerDrain(this.agentIdentifier, this.featureName)
+        this.auto = true
+        this.importAggregator()
```

Comment:
We probably need to add a check to `importAggregator()` that early returns if its already staged (but has not actually imported the agg), so it doesnt set up two window listeners

Reply:
importAggregator still needs a 
`if (!this.auto) return // instrument file does not import aggregate if feat waits for start API`

Reply:
the `startedAlready` flag can be inside this cb listener in Jordan's case